Source/Core/Core/AchievementManager.h

```diff
@@ -61,7 +70,21 @@ class AchievementManager
   rc_api_fetch_game_data_response_t m_game_data{};
   bool m_is_game_loaded = false;
 
+  struct UnlockStatus
+  {
+    unsigned game_data_index;
+    enum class UnlockType
+    {
+      LOCKED,
+      SOFTCORE,
+      HARDCORE
+    } remote_unlock_status = UnlockType::LOCKED;
+    int session_unlock_count = 0;
+  };
+  std::map<int /*achievement ID*/, UnlockStatus> m_unlock_map;
```

Comment:
Maybe just have a:

```cpp
using AchievementID = int;
```

alias defined somewhere and use it here, so the comment isn't necessary.